PythonVisualizations/BinaryTreeBase.py

```diff
@@ -0,0 +1,418 @@
+from tkinter import *
+import time, random
+from enum import Enum
+
+try:
+    from drawable import *
+    from VisualizationApp import *
+except ModuleNotFoundError:
+    from .drawable import *
+    from .VisualizationApp import *
+
+class Child(Enum):
+    LEFT = 0
+    RIGHT = 1
+
+class Node(object):
+   def __init__(self, drawable, coords, tag):
+      self.drawable = drawable
+      self.coords = coords
+      self.tag = tag
+
+   def getKey(self):
+      return self.drawable.val
+
+   def setKey(self, k):
+      self.drawable.val = k
+
+   def __str__(self):
+      return "{" + str(self.getKey()) + "}"
+
+class BinaryTreeBase(VisualizationApp):
+   # -------- CONSTANTS ------------------
+   FONT_SIZE = '16'
+   VALUE_FONT = ('Helvetica', FONT_SIZE)
+   FOUND_FONT = ('Helvetica', FONT_SIZE)
+
+   WIDTH = 1000
+   HEIGHT = 400
+
+   CIRCLE_SIZE = 15
+   NODE_X_GAP = 400
+   NODE_Y_GAP = 75
+   ROOT_X0 = WIDTH/2
+   ROOT_Y0 = CIRCLE_SIZE + 5
+   MAX_LEVEL = 5
+
+   ARROW_HEIGHT = 30
+
+   def __init__(self, title="Tree", width = WIDTH, height = HEIGHT, **kwargs):
+      super().__init__(title=title, canvasWidth= width, canvasHeight= height, **kwargs)
```

Comment:
I would like for the tree display to be more customizable by the caller.  For example, the BinarySearchTree visualization would put the tree in the center of the canvas, while the Heap would shift it to the right so it could show the array representation of the tree on the left of the canvas.   So I think the location of the root, the circle size, the distance between levels, and the width of the tree at the MAX_LEVEL, should all be parameters to the constructor.   The `canvasWidth` and `canvasHeight` should not be parameters since they are managed by the VisualizationApp and can be passed via the `kwargs` parameter.

Reply:
These changes were implemented in commit 621766b